src/PackageJsonSynchronizer.php

```diff
@@ -150,6 +149,43 @@ private function registerWebpackResources(array $phpPackages)
         file_put_contents($controllersJsonPath, json_encode($newControllersJson, \JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES)."\n");
     }
 
+    public function registerPeerDependencies(array $phpPackages)
+    {
+        $peerDependencies = [];
+
+        foreach ($phpPackages as $phpPackage) {
+            if (!$assetsDir = $this->resolveAssetsDir($phpPackage)) {
+                continue;
+            }
+
+            $packageJsonPath = $this->rootDir.'/vendor/'.$phpPackage.$assetsDir.'/package.json';
+            $packageJson = (new JsonFile($packageJsonPath))->read();
+
+            foreach ($packageJson['peerDependencies'] ?? [] as $peerDependency => $constraint) {
+                $peerDependencies[$peerDependency][$constraint] = $constraint;
+            }
+        }
+
+        if (!$peerDependencies) {
+            return;
+        }
+
+        $manipulator = new JsonManipulator(file_get_contents($this->rootDir.'/package.json'));
+        $content = json_decode($manipulator->getContents(), true);
+        $devDependencies = $content['devDependencies'];
+
+        foreach ($peerDependencies as $peerDependency => $constraints) {
+            if (!isset($content['dependencies'][$peerDependency]) && !isset($content['devDependencies'][$peerDependency])) {
+                // FIXME: we should compute the intersection of all constraints, including any pre-existing one
```

Comment:
this should not be a blocker for this PR (as this is hard to implement and is not mandatory to provide a nice DX improvement)